test/jdk/java/awt/TitledBorder/ScaledEtchedBorderTest.java

```diff
@@ -0,0 +1,248 @@
+            } else if (checkHighlight) {
+                if (color == shadow.getRGB()) {
+                    thickness++;
+                } else if (color == highlight.getRGB()) {
+                    verifyThickness(x, y, thickness, scaling, "Vertical");
+                    checkHighlight = false;
+                    checkShadow = true;
+                    thickness = 1;
+                } else {
+                    throw new RuntimeException("Vertical Border has empty space between highlight and shadow.");
+                }
+            } else {
+                if (color == shadow.getRGB()) {
+                    throw new RuntimeException("Border colors reversed.");
+                } else if (color == highlight.getRGB()) {
+                    thickness++;
+                } else {
+                    verifyThickness(x, y, thickness, scaling, "Vertical");
+                    checkShadow = false;
+                    thickness = 0;
+                }
+            }
+        }
+    }
+
+    private static void createGUI(boolean showFrame) {
+        // Render content panel
+        contentPanel = new JPanel();
+        contentPanel.setLayout(new BoxLayout(contentPanel, BoxLayout.Y_AXIS));
+
+        for (int i = 0; i < 4; i++) {
+            parentPanel = new JPanel(new BorderLayout());
+            parentPanel.setBorder(BorderFactory.createEmptyBorder(0, i, 4, 4));
+
+            childPanel = new JPanel(new BorderLayout());
+            childPanel.setBorder(BorderFactory.createEtchedBorder(highlight, shadow));
+            childPanel.add(Box.createRigidArea(SIZE), BorderLayout.CENTER);
+
+            parentPanel.add(childPanel, BorderLayout.CENTER);
+            contentPanel.add(parentPanel);
+        }
+
+        frame = new JFrame("Swing Test");
+        frame.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+        frame.getContentPane().add(contentPanel, BorderLayout.CENTER);
+        frame.pack();
+        frame.setLocationRelativeTo(null);
+
+        for (double scaling : scales) {
+            // Create BufferedImage
+            BufferedImage buff = new BufferedImage((int) Math.ceil(contentPanel.getWidth() * scaling),
+                    (int) Math.ceil(contentPanel.getHeight() * scaling),
+                    BufferedImage.TYPE_INT_ARGB);
+            Graphics2D graph = buff.createGraphics();
+            graph.scale(scaling, scaling);
+            // Painting panel onto BufferedImage
+            contentPanel.paint(graph);
+            graph.dispose();
+            // Save each image ? -- Here it's useful for debugging
+            saveImage(buff, String.format("test%.2f.png", scaling));
+            images.add(buff);
+        }
+        // Save coordinates of the panels
+        for (Component comp : contentPanel.getComponents()) {
+            panelLocations.add(comp.getLocation());
+        }
+
+        if (showFrame) {
+            frame.setVisible(true);
+        } else {
+            frame.dispose();
+        }
```

Comment:
That looks odd. I mean if you show the frame, how is it disposed ?

And more broadly ... why do you even create the frame if you never show it ?
And if you don't create the frame, this test can be headless, can't it ?

Reply:
Perhaps you are trying to get layout to happen ? That should be possible but it might be tricky.
But I think if you are going so far as pack() and can't change that, then show the frame regardless.

Reply:
Right, JPanel isn't laid out without being added to a peered component.

Maybe setting explicit positions to child panels will work, then the test can be headless.

Reply:
> That looks odd. I mean if you show the frame, how is it disposed ?

The idea for that is that frame is shown only if `-show` parameter is passed. In this case, it's assumed the user runs the test from the IDE or on command line. So the frame is disposed of when the user closes it.

Reply:
> > Perhaps you are trying to get layout to happen ? That should be possible but it might be tricky.
>
> Right, JPanel isn't laid out without being added to a peered component.
> 
> Maybe setting explicit positions to child panels will work, then the test can be headless.

Indeed, it works with explicitly set size and positions. Here's [a working sample](https://github.com/alisenchung/jdk/compare/8279614...aivanov-jdk:alisen-8279614-titledBorder). (Add `.diff` or `.patch` to the URL to get the changeset to apply.)